mergify_cli/ci_issues/junit_upload.py

```diff
@@ -0,0 +1,63 @@
+import sys
+
+import httpx
+import pydantic
+
+from mergify_cli import console
+from mergify_cli.ci_issues import utils
+from mergify_cli.ci_issues.custom_types import CIIssuesJunitUploadParameters
+
+
+async def upload(  # noqa: PLR0913, PLR0917
+    api_url: str,
+    token: str,
+    repository: str,
+    files: str,
+    head_sha: str,
+    job_name: str,
+    provider: str | None,
+) -> None:
+    try:
+        # NOTE(leo): perform a first validation of the input parameters in CLI
+        # before the back-end handles the rest
```

Comment:
So this is a bad idea in general, because if we change the backend you also need to change the CLI. It should not try to do what's not its job here in general.